### DISCUSSION TITLE
Is there a way to get the citekey via the local Zotero API?

### DISCUSSION THREAD

**Discussion by jonasengelmann** [retorquere/zotero-better-bibtex#3034] [Q&A] [answered]
I usually format quotations using Zotero Links in my documents, so that I can jump back to the relevant passage in the PDF. This all works great, but now I want to finish a document and replace those links automatically with citekeys. Is there a way to retrieve the citekey of a specific Zotero item of which I only have the Zotero ID? I tried using the new local Zotero API under: http://localhost:23119/api/users/0/items/5CUD43D6 but the citekey is not part of the response.

**Comment by retorquere** [comment, marked as answer]
not possible until the real citation key field comes to zotero, but you can get them through BBT JSON-RPC:

```
curl http://localhost:23119/better-bibtex/json-rpc -X POST -H "Content-Type: application/json" -H "Accept: application/json" --data-binary '{"jsonrpc": "2.0", "method": "item.citationkey", "params": {"item_keys": ["shelah1970c"]}}'
```

**Reply by jonasengelmann** [reply]
Thank you so much! However, I always get a null result back. Am I right to use as item_key the Zotero-ID (5CUD43D6 from the above example)?

**Reply by jonasengelmann** [reply]
I am sorry, I was wrong, I used a Zotero attachment key, instead of an item key, now everything works! Thank you again!
